src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -1075,6 +1057,67 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
         }
     }
 
+    /// <summary>
+    /// Handles the validation of source and target fields as valid backing columns in the DB.
+    /// We put the logic for both source and target fields into a single function here for readability
+    /// when catching exceptions in the caller related to being unable to lookup the backing columns.
+    /// This happens when validation is called with a mal formed connection string.
+    /// </summary>
+    /// <param name="invalidColumns">List in which to aggregate the invalid fields.</param>
+    /// <param name="sourceFields">List of the backing source fields to check for existence in backing DB.</param>
+    /// <param name="targetFields">List of the backing target fields to check for existence in backing DB.</param>
+    /// <param name="relationshipName">Name of the relationship.</param>
+    /// <param name="entityName">The name of the entity that we check for backing columns.</param>
+    /// <param name="targetEntityName">Name of the target entity.</param>
+    /// <param name="sqlMetadataProvider">The sqlMetadataProvider used to lookup if the backing fields are valid columns in DB.</param>
+    private void ValidateSourceAndTargetFieldsAsBackingColumns(
+        List<string> invalidColumns,
+        string[]? sourceFields,
+        string[]? targetFields,
+        string relationshipName,
+        string entityName,
+        string targetEntityName,
+        ISqlMetadataProvider sqlMetadataProvider)
+    {
+        // In all kinds of relationships, if sourceFields are included they must be valid columns in the backend.
+        if (sourceFields is not null)
```

Comment:
is there another place in validation code that checks whether the targetEntity is valid (metadata acquired too)? 
- perhaps you should do that here, there are a few datastructures that can give you this detail.
- Fail early with a precise error message that allows you to leave SqlMetadataProvider as-is to allow for refactoring later.

Reply:
We do indeed have validation code for checking the metadata (which in this case is stored as a DatabaseObject) on both the source and target entities in the outer `ValidateRelationshipsInConfig` function.